modules/ldap-auto-syncing.adoc

```diff
@@ -0,0 +1,246 @@
+            pageSize: 0
+        groupUIDAttribute: dn
+        groupNameAttributes: [ cn ]
+        groupMembershipAttributes: [ member ]
+        usersQuery:
+            baseDN: "ou=users,dc=example,dc=com" <5>
+            scope: sub
+            derefAliases: never
+            pageSize: 0
+        userUIDAttribute: dn
+        userNameAttributes: [ uid ]
+        tolerateMemberNotFoundErrors: false
+        tolerateMemberOutOfScopeErrors: false
+----
+<1> Define the desired LDAP config. This example uses the RFC2307 schema; adjust values as necessary.
+<2> Edit the url as required.
+<3> Edit the `bindDN` as required.
+<4> Edit the groupsQuery `baseDN` as required.
+<5> Edit the usersQuery `baseDN` as required.
++
+// TODO: point out additional fields as necessary. Also work on better callouts.
+
+. Create the config map:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-config-map.yaml
+----
+
+. Define a cron job:
++
+.Example `ldap-sync-cron-job.yaml`
+[source,yaml]
+----
+kind: CronJob
+apiVersion: batch/v1beta1
+metadata:
+  name: ldap-group-syncer
+  namespace: ldap-sync
+  labels:
+    app: cronjob-ldap-group-sync
+spec:
+  schedule: "*/10 * * * *" <1>
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: 5
+  failedJobsHistoryLimit: 5
+  jobTemplate:
+    metadata:
+      labels:
+        app: cronjob-ldap-group-sync
+    spec:
+      backoffLimit: 0
+      template:
+        metadata:
+          labels:
+            app: cronjob-ldap-group-sync
+        spec:
+          containers:
+            - name: ldap-group-sync
+              image: "openshift/origin-cli:latest"
+              command:
+                - "/bin/bash"
+                - "-c"
+                - oc adm groups sync --sync-config=/etc/config/ldap-group-sync.yaml --confirm
+              volumeMounts:
+                - mountPath: "/etc/config"
+                  name: "ldap-sync-volume"
+                - mountPath: "/etc/secrets"
+                  name: "ldap-bind-password"
+                - mountPath: "/etc/ldap-ca"
+                  name: "ldap-ca"
+          volumes:
+            - name: "ldap-sync-volume"
+              configMap:
+                name: "ldap-group-syncer"
+            - name: "ldap-bind-password"
+              secret:
+                secretName: "v4-0-config-user-idp-0-bind-password" <2>
+            - name: "ldap-ca"
+              configMap:
+                name: "v4-0-config-user-idp-0-ca" <3>
+          restartPolicy: "Never"
+          terminationGracePeriodSeconds: 30
+          activeDeadlineSeconds: 500
+          dnsPolicy: "ClusterFirst"
+          serviceAccountName: "ldap-group-syncer"
+          serviceAccount: "ldap-group-syncer"
+----
+<1> Schedule for the job specified in link:https://en.wikipedia.org/wiki/Cron[cron format]. In this example, the cron job runs every 10 minutes.
+<2> This secret was created when the LDAP IDP was configured. TODO: Should this just be "ldap-secret" - per the LDAP IDP docs?
```

Comment:
@stlaz Prior to this, we said that you had to set up the LDAP IDP. If you look at our docs on it [1] the names are `ldap-secret` for the secret, and `ca-config-map`. Any thoughts on if we should use those names, or are they too generic? 

[1] https://docs.openshift.com/container-platform/4.8/authentication/identity_providers/configuring-ldap-identity-provider.html#identity-provider-creating-ldap-secret_configuring-ldap-identity-provider

Reply:
@stlaz Did you have any thoughts on this question here?

Reply:
I think you can refer to the secret used in the IdP configuration, specific names are probably not necessary in the description.